governance/program/src/state/proposal.rs

```diff
@@ -583,6 +647,11 @@ impl ProposalV2 {
         Ok(max_voter_weight)
     }
 
+    /// Makes sum of weights of all votes that have been voted at proposal until now
+    pub fn resolve_all_votes_weight(&self) -> u64 {
```

Comment:
This is incorrect, because in case where users can place their vote weight on multiple options the same weight would be calculated multiple times.
You have to introduce a field on `Proposal` account to track the total weight of approving votes. You can use the space from `reserved: [u8; 64]` to introduce the field